stdlib/LinearAlgebra/src/eigen.jl

```diff
@@ -647,9 +647,34 @@ function Base.isequal(A::Eigen, B::Eigen)
 end
 
 # Conversion methods
-
 ## Can we determine the source/result is Real?  This is not stored in the type Eigen
-AbstractMatrix(F::Eigen) = F.vectors * Diagonal(F.values) / F.vectors
+AbstractMatrix(F::Eigen) = F.are_vectors_unitary ?
+                             (F.vectors * Diagonal(F.values) * (F.vectors')) :
+                             (F.vectors * Diagonal(F.values) / F.vectors)
 AbstractArray(F::Eigen) = AbstractMatrix(F)
 Matrix(F::Eigen) = Array(AbstractArray(F))
 Array(F::Eigen) = Matrix(F)
+
+# Operations on Eigen
+# Only operations that can be efficiently implemented or conserve the eigen type are implemented
+adjoint(E::Eigen) = Eigen(adjoint.(E.values),
+                          copy(adjoint(E.are_vectors_unitary ? (E.vectors') : inv(E.vectors))),
+                          E.are_vectors_unitary)
+transpose(E::Eigen) = Eigen(transpose.(E.values),
+                            copy(transpose(E.are_vectors_unitary ? (E.vectors') : inv(E.vectors))),
+                            E.are_vectors_unitary && (eltype(E) <: Real))
+
+det(A::Eigen) = prod(A.values)
+tr(E::Eigen) = sum(E.values)
+
+*(E::Eigen, x::AbstractVector) = E.vectors * (E.values .* (E.are_vectors_unitary ? (E.vectors'x) : (E.vectors \ x)))
+*(x::Adjoint{<:Any, <:AbstractVector}, E::Eigen) = E.are_vectors_unitary ?
+                                                   ((x*(E.vectors)) * Diagonal(E.values)) * (E.vectors') :
+                                                   ((x*(E.vectors)) * Diagonal(E.values)) / E.vectors
+mul!(x, E::Eigen, y::AbstractVector) = copyto!(x, E*y)
```

Comment:
What's the point of providing this method if it doesn't provide any improvement over `E*y`(and makes it actually even worse, against expectations)?

Reply:
Very reasonable point, I'll remove it!